### PR TITLE
🐛 FH-3197 Changes to support running tests agains sync scaling changes

### PR DESCRIPTION
**Pull request by david-martin**
* Use docker for running mongodb to allow common testing prereqs
* Setup a mongodb test user to trigger 'upgraded db' mode in fh-db
* Explicitly specify the mongo connection url for the test server
* Use the branch of fh-mbaas-api with sync scaling changes
* Fix an issue with sync.manage to return a Promise
* Allow for slightly different behaviour when stopping & forcing sync. Specificially, do a second force sync to ensure our updates were applied.

### CODE REVIEW

**Comment by david-martin** [comment]
@aidenkeating Note there is 1 test that will not pass for both old & new sync code 
`should not stop remote updates using forceSync while sync not active`
This is because of the change in sync server to accept updates, put them on a queue, and respond immediately.
Because of this, we need to do 2 force syncs.

File: package.json
Comment (aidenkeating) [inline package.json]: Is the plan to keep this branch here until it's merged into master and do these tests fail against master?

If so, we should probably disable these tests running by default against `fh-mbaas-api` master.
https://github.com/feedhenry/fh-mbaas-api/blob/master/.travis.yml#L23

Also, I don't see these tests being run against the horizontal scaling branch, maybe we should include these being run there now that they're more stable? What do you think?

**Review by aidenkeating — APPROVED** [review]
Looks good to me 👍 One question down below or up above, I can't remember how GitHub formats reviews.

**Comment by david-martin** [comment]
@aidenkeating Updated the PR to remove the specific branch of fh-mbaas-api.
Merging this now to unblock changes to the fh-mbaas-api in https://github.com/feedhenry/fh-mbaas-api/pull/96 to run acceptance tests for CI builds there

**Pull request merged**
